UefiPayloadPkg/Library/PciSegmentInfoLibAcpiBoardInfo/PciSegmentInfoLibAcpiBoardInfo.c

```diff
@@ -67,6 +67,11 @@ RetrieveMultiSegmentInfoFromHob (
     mPciSegments[Index].SegmentNumber  = (UINT16)(PciRootBridgeInfo->RootBridge[Index].Segment);
     mPciSegments[Index].StartBusNumber = (UINT8)PciRootBridgeInfo->RootBridge[Index].Bus.Base;
     mPciSegments[Index].EndBusNumber   = (UINT8)PciRootBridgeInfo->RootBridge[Index].Bus.Limit;

```

Comment:
Here we need additional new logic like below:

After all the mPciSegments elements initialized, we need to compare and consolidate the duplicate SegmentNumber elements.
If SegmentNumber is the same, we will update first-found mPciSegments->StartBusNumber and EndBusNumber and ignore rest of the others.
The NumberOfRootBridges is actually "PciSegmentInfoCount"(I think we can rename it), so if we have consolidated the elements we should reduce the PciSegmentInfoCount.

Reply:
will handle this in separate PR.